docworks-cli/src/docworks.js

```diff
@@ -50,49 +46,52 @@ function docworks() {
   /* eslint-enable no-console */
 
   function ecp() {
-    let argv = optimist
-      .usage('Usage: $0 ecp -r [remote repo] [-b [remote branch]] -s [local sources] -p [file pattern] [-ed [enrichment docs directory]] [--plug [plugin]] [--dryrun]')
-      .demand('r')
+    const commandConfig = optimist
+      .usage('Usage: $0 ecp -r [remote repo] [-b [remote branch]] -fs [local sources] -fp [file pattern] [-ed [enrichment docs directory]] [--plug [plugin]] [--dryrun]')
       .alias('r', 'remote')
       .describe('r', 'remote repository to merge docs into')
       .alias('b', 'branch')
       .describe('b', 'branch on the remote repository to work with')
-      .demand('fs')
       .alias('fs', 'sources')
       .describe('fs', 'one or more folders containing the source files to extract docs from')
       .alias('fx', 'excludes')
       .describe('fx', 'one or more folders to exclude (including their children) from extracting docs')
-      .default('fp', '.+\\.js?$')
       .alias('fp', 'pattern')
       .describe('fp', 'file pattern, defaults to ".+\\.js$"')
-      .demand('p')
       .alias('p', 'project')
       .describe('p', 'project folder name in the docs repo')
       .describe('ed', 'project enrichment docs relative directory')
       .describe('plug', 'a module name that is a jsdoc or docworks plugin')
       .describe('dryrun', 'dry run - do not push to remote repo')
-      .parse(process.argv.slice(3))
+      .describe('config', 'file to load configurations from')
 
-    let remote = argv.remote
-    let branch = argv.branch
-    let sources = argv.sources
-    let excludes = argv.excludes ? (Array.isArray(argv.excludes) ? argv.excludes : [argv.excludes]) : []
-    let pattern = argv.pattern
-    let project = argv.project
-    let dryrun = !!argv.dryrun
-    let enrichmentDocsDir = argv.ed
-    let plugins = resolveAndInitPlugins(argv.plug)
+    const argv = commandConfig.parse(cliArgs)
+
+    const config = argv.config
+      ? Object.assign(require(argv.config), argv)
+      : argv
+
+    if (!config.remote || !config.sources || !config.project) {
+      commandConfig.showHelp()
+      process.exit(1)
+    }
 
-    tmp.dir().then(o => {
+    return tmp
+      .dir()
+      .then(({ path: tmpWorkingDir }) => {
       return extractComparePush({
-        remoteRepo: remote,
-        remoteBranch: branch,
-        workingDir: o.path,
-        projectSubdir: project,
-        jsDocSources: {'include': sources, 'includePattern': pattern, 'exclude': excludes},
-        plugins,
-        enrichmentDocsDir,
-        dryrun
+          remoteRepo: config.remote,
+          remoteBranch: config.branch,
+          workingDir: tmpWorkingDir,
+          projectSubdir: config.project,
+          jsDocSources: {
+            include: config.sources,
+            includePattern: config.pattern || '.+\\.js?$',
```

Comment:
I now understand why you removed the default from the `fp` argument, but I still think it's pretty confusing in the command description. Leaving the comments in place as it made me confused, but I can see why you'd want to keep it as is - think about it.